src/dmd/aggregate.d

```diff
@@ -372,16 +370,27 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
                 if (!MODimplicitConv(v2.type.mod, vd.type.mod))
                     vd.overlapUnsafe = true;
 
-                if (!vx)
+                if (i > j)
                     continue;
-                if (v2._init && v2._init.isVoidInitializer())
+
+                if (!v2._init)
                     continue;
 
-                if (vx._init && v2._init)
+                if (v2._init.isVoidInitializer())
+                    continue;
+
+                if (vd._init && !vdIsVoidInit && v2._init)
                 {
                     .error(loc, "overlapping default initialization for field `%s` and `%s`", v2.toChars(), vd.toChars());
                     errors = true;
                 }
+                else if (v2._init && i < j)
+                {
+                    // @@@DEPRECATED_v2.086@@@.
```

Comment:
See https://github.com/dlang/DIPs/blob/master/DIPs/accepted/DIP1013.md  for the specification on how these deprecations work.

I still don't understand it, but my the convention I've been following is that this should state the version for the next step in the deprecation process.  So, this feature will be deprecated in 2.088.  The deprecation will last for 1 year and we release every 2 months, so that would make the next step 88 + 6 = 2.094.  So the comment should read `@@@DEPRECATED_v2.094@@@`.  I believe the reason for this is that after a release, someone may want to grep for any deprecations that need to move forward for that release.  However, I must admit, I still don't understand it.  That is why I recommend the following:

```
// @@@DEPRECATED_v2.094@@@
// Deprecated in 2.088
// Make an error in 2.094
// Remove from the language in 2.100
```
cc @wilzbach